script.js

```diff
@@ -1,4 +1,273 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+// ❌⚔🤺
+
+const DEFAULT_SETTINGS = {
+  highScoreMode: false,
+  knockoutMode: false,
+  numPlayers: 10,
+  numDice: 2,
+  round: 1,
+  players: [],      // used for knockout mode, players not yet eliminated
+  pointerX: -1,     // used for knockout mode, index of playerX
+  pointerY: -1,     // used for knockout mode, index of playerY
+  playerRolls: [],  // knockout mode - stores rolls and scores of a round
+                    // regular mode - stores scores of a round
+  playerScores: [], // used for regular mode, cumulative scores of players
+}
+
+var S = {...DEFAULT_SETTINGS}
```

Comment:
nice way to store the default state and make a copy of it to be your state object. the variable S can be more descriptive